sql/core/src/main/scala/org/apache/spark/sql/util/QueryExecutionListener.scala

```diff
@@ -74,15 +75,19 @@ trait QueryExecutionListener {
 // The `loadExtensions` flag is used to indicate whether we should load the pre-defined,
 // user-specified listeners during construction. We should not do it when cloning this listener
 // manager, as we will copy all listeners to the cloned listener manager.
-class ExecutionListenerManager private[sql](session: SparkSession, loadExtensions: Boolean)
+class ExecutionListenerManager private[sql](
+    session: SparkSession,
+    sqlConf: SQLConf,
+    loadExtensions: Boolean)
   extends Logging {
 
   private val listenerBus = new ExecutionListenerBus(this, session)
 
   if (loadExtensions) {
     val conf = session.sparkContext.conf
     conf.get(QUERY_EXECUTION_LISTENERS).foreach { classNames =>
-      Utils.loadExtensions(classOf[QueryExecutionListener], classNames, conf).foreach(register)
```

Comment:
what if we wrap this with `SQLConf.withExistingConf(conf) {}`? Then people can do `SQLConf.get` in their listeners.

Reply:
> what if we wrap this with `SQLConf.withExistingConf(conf) {}`? Then people can do `SQLConf.get` in their listeners.

Here the `conf` is a SparkConf, can't use SQLConf.withExistingConf, also why I use `sqlConf`  passed from `BaseSessionStateBuilder` is aimed to use same sql conf in current SessionState. Since `ExecutionListenerManager` belongs to current session.

Reply:
sorry, `SQLConf.withExistingConf(sqlConf) {}`.

I'm fine to pass sql conf to `ExecutionListenerManager`, as it's an internal class. I'm against changing the constructor of `QueryExecutionListener`, as it's a breaking change.

Reply:
> sorry, `SQLConf.withExistingConf(sqlConf) {}`.
> 
> I'm fine to pass sql conf to `ExecutionListenerManager`, as it's an internal class. I'm against changing the constructor of `QueryExecutionListener`, as it's a breaking change.

But it won't break any old usage. Can be a new feature?

Reply:
I see, the new code has a fallback to the old constructor. But isn't `SQLConf.withExistingConf(sqlConf)` a much simpler solution?

Reply:
> I see, the new code has a fallback to the old constructor. But isn't `SQLConf.withExistingConf(sqlConf)` a much simpler solution?

Yea, updated